### PR TITLE
Add id to testcase name for secrets to prevent deduplication

### PR DESCRIPTION
**Pull request by Urth**
This issue was [fixed for the sast parser](https://github.com/angrymeir/SecScanner2JUnit/issues/9) but not for the secrets.
- Add id to testcase name for secrets
- Use vulnerability class to parse findings
- Add test for secrets

### CODE REVIEW

**Comment by angrymeir** [comment]
Hey @Urth, first of all thank you for the effort you put in this pull request! I appreciate your work :)

Unfortunately atm I'm quite busy with other projects. If this is an urgent matter, I'd suggest we rather go for the solution from #9.
Reason is that I'd like to first implement the `SAST` and `Secrets` cases according to the scheme @marwin1991 proposed and implemented for `Container Scanning` in #23 so that the codebase stays consistent (well tbh it's not consistent atm since I did not adjust `SAST`/`Secrets` yet, so rather that it doesn't drift any further...)

If that's fine for you, I'll add the `id` to the testcase name today/tomorrow.

Rewrite of `SAST` and `Secret` is planned for somewhere between 26.12. and 31.12.

**Comment by Urth** [comment]
I'm in no hurry but I should point out that my changes are based on the container implementation and use the classes that were introduced with https://github.com/angrymeir/SecScanner2JUnit/pull/23
So this should be most of the secrets rewrite completed.
If you want to make additional changes just let me know.

File: secscanner2junit/vulnerability.py [hunk 1]
```diff
@@ -105,19 +107,16 @@ def get_severity(self):
 class SastVulnerability(_Vulnerability):
```
Comment (marwin1991) [inline secscanner2junit/vulnerability.py:107]: Maybe we can create `SecrectsVulnerability(_Vulnerability):` ?

**Comment by marwin1991** [comment]
@Urth Really good job!

File: secscanner2junit/vulnerability.py (hunk 1 above)
Comment (Urth) [reply secscanner2junit/vulnerability.py:107]: The secrets output is compatible with sast so I chose not to do this. I could subclass the SastVulnerability or introduce a new SastBaseVulnerability. And do the same for the SastLocation? It felt like adding duplication so I was on the fence about this. If you have a strong opinion on this I'm happy to make the change.

Comment (marwin1991) [reply secscanner2junit/vulnerability.py:107]: it was based on https://gitlab.com/gitlab-org/security-products/security-report-schemas/-/tree/master/src

Both of this report are based on https://gitlab.com/gitlab-org/security-products/security-report-schemas/-/blob/master/src/sast-report-format.json

So maybe your idea is correct :)

We could model classes to be comatible with this schemas what will be useful to others?

@Urth What do you think?

**Pull request closed without merging**